ui/admin/tests/e2e/helpers/boundary-cli.js

```diff
@@ -66,18 +67,471 @@ exports.connectToTarget = async (targetId) => {
+         -scope-id=global \
+         -format json`
+      )
+    ).item;
+  } catch (e) {
+    console.log(`${e.stderr}`);
+  }
+  return newOrg.id;
+};
+
 /**
  * Uses the boundary CLI to delete the specified organization
  * @param {string} orgId ID of the organization to be deleted
  */
-exports.deleteOrg = async (orgId) => {
+exports.deleteOrgCli = async (orgId) => {
   try {
     exec('boundary scopes delete -id=' + orgId);
   } catch (e) {
     console.log(`${e.stderr}`);
   }
 };
 
+/**
+ * Uses the boundary CLI to create a new project.
+ * @param {string} orgId ID of the organization under which the project will be created.
+ * @returns {string} new project's ID
+ */
+exports.createNewProjectCli = async (orgId) => {
+  const projectName = 'Project ' + nanoid();
+  let newProject;
+  try {
+    newProject = JSON.parse(
+      execSync(
+        `boundary scopes create \
+         -name="${projectName}" \
+         -scope-id=${orgId} \
+         -format json`
+      )
+    ).item;
+  } catch (e) {
+    console.log(`${e.stderr}`);
+  }
+  return newProject.id;
+};
+
+/**
+ * Uses the boundary CLI to create a new controller-led worker.
+ * @returns {string} new worker's ID
+ */
+exports.createNewControllerLedWorkerCli = async () => {
+  const workerName = 'worker-' + nanoid();
+  let newWorker;
+  try {
+    newWorker = JSON.parse(
+      execSync(
+        `boundary workers create controller-led \
+         -name="${workerName.toLowerCase()}" \
+         -format json`
+      )
+    ).item;
+  } catch (e) {
+    console.log(`${e.stderr}`);
+  }
+  return newWorker.id;
+};
+
+/**
+ * Uses the boundary CLI to create a new password auth-method.
+ * @param {string} scopeId ID of the scope under which the auth-method will be created.
+ * @returns {string} new auth-method's ID
+ */
+exports.createNewPasswordAuthMethodCli = async (scopeId) => {
+  const authMethodName = 'auth-method-' + nanoid();
+  let newAuthMethod;
+  try {
+    newAuthMethod = JSON.parse(
+      execSync(
+        `boundary auth-methods create password \
+         -name "${authMethodName}" \
+         -scope-id ${scopeId} \
+         -format json`
+      )
+    ).item;
+  } catch (e) {
+    console.log(`${e.stderr}`);
+  }
+  return newAuthMethod.id;
+};
+
+/**
+ * Uses the boundary CLI to make an auth-method primary.
+ * @param {string} scopeId ID of the scope that the auth-method belongs to.
+ * @param {string} authMethodId ID of the auth method that will be made primary.
+ * @returns {object} make auth method primary response.
+ */
+exports.makeAuthMethodPrimaryCli = async (scopeId, authMethodId) => {
+  let makeAuthMethodPrimaryResponse;
+  try {
+    makeAuthMethodPrimaryResponse = JSON.parse(
+      execSync(
+        `boundary scopes update \
+         -id=${scopeId} \
+         -primary-auth-method-id ${authMethodId} \
+         -format json`
+      )
+    ).item;
+  } catch (e) {
+    console.log(`${e.stderr}`);
+  }
+  return makeAuthMethodPrimaryResponse;
```

Comment:
Wonder if we should change this to return a `bool` on whether or not it's successful? Not sure if we really want to add logic about interpreting CLI responses into tests.

Reply:
Discussed with @moduli. For now, not returning anything from the method. 
If there is an error in the CLI command used in this method, it will be caught in the catch block and logged. 
The test will fail on a remove primary auth-method attempt.